lightning/cogs/automod/ui.py

```diff
@@ -191,3 +203,338 @@ async def warn_ban(self, itx: discord.Interaction, button: discord.ui.Button):
         self.choice = "warn_ban"
         await itx.response.edit_message(view=None)
         self.stop()
+
+
+class FakeCtx:
+    def __init__(self, member) -> None:
+        self.author = member
+
+
+class GatekeeperRoleView(BasicMenuLikeView):
+    @discord.ui.select(cls=discord.ui.RoleSelect, placeholder="Select a role", min_values=1, max_values=1)
+    async def callback(self, itx: discord.Interaction[LightningBot], select: discord.ui.RoleSelect):
+        assert itx.guild is not None
+
+        role = select.values[0]
+
+        if role >= itx.guild.me.top_role:
+            await itx.response.send_message("You cannot use this role because it is higher than my role!",
+                                            ephemeral=True)
+            return
+
+        if has_dangerous_permissions(role.permissions):
+            await itx.response.send_message("You cannot use this role because it contains permissions that are deemed"
+                                            " dangerous!", ephemeral=True)
+            return
+
+        await self.insert_role(itx, role)
+        await itx.response.send_message(f"Set {role.name} ({role.mention}) as the gatekeeper role!")
+        self.stop(interaction=itx)
+
+    async def insert_role(self, interaction: discord.Interaction[LightningBot], role: discord.Role):
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (docstrings):** Please update the docstring for function: `insert_role`

Reason for update: The function signature has changed to include the interaction parameter.

Suggested new docstring:
```
Inserts the selected role into the database using the provided interaction to access the client's pool.
```